api/src/main/java/org/pmiops/workbench/db/dao/UserService.java

```diff
@@ -71,21 +72,21 @@ DbUser syncComplianceTrainingStatusV1()
       throws org.pmiops.workbench.moodle.ApiException, NotFoundException;
 
   @Deprecated
-  DbUser syncComplianceTrainingStatusV1(DbUser user)
+  DbUser syncComplianceTrainingStatusV1(DbUser user, Agent agent)
       throws org.pmiops.workbench.moodle.ApiException, NotFoundException;
 
   DbUser syncComplianceTrainingStatusV2()
       throws org.pmiops.workbench.moodle.ApiException, NotFoundException;
 
-  DbUser syncComplianceTrainingStatusV2(DbUser user)
+  DbUser syncComplianceTrainingStatusV2(DbUser user, Agent agent)
```

Comment:
This is much cleaner, thanks. 

Only thing that would be neater would be to inject the agent somehow, but I'm guessing that's not trivial because of threading and auth concerns.

Reply:
It's not trivial because it depends on service-level business logic. There is nothing inherent to a user identity which tells us whether they are acting as an admin or a user. Possibly this is something we could bind in an interceptor, based on what method is being called, but being explicit in controller methods is probably fine for now..